src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,161 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_validUser() {
+        User validUser = new User();
+        validUser.setAge(20);
+        validUser.setPassword("passwordValid");
+        validUser.setLogin("loginValid");
+
+        registrationService.register(validUser);
+        assertNotNull(storageDao.get("loginValid"));
+        assertEquals(validUser, storageDao.get("loginValid"));
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        User firstUser = new User();
+        firstUser.setLogin("validLogin");
+        firstUser.setPassword("validPassword");
+        firstUser.setAge(20);
+        storageDao.add(firstUser);
```

Comment:
You should not use the `add` method of the `StorageDao` to add users in your tests. This method is part of the system under test and its correctness is not guaranteed. Instead, you should directly manipulate the `Storage.people` list to prepare your test data.